src/Test.UnitTests.Sarif.Multitool/QueryCommandPropertyBagTests.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using FluentAssertions;
+using Microsoft.CodeAnalysis.Sarif.Query;
+using Microsoft.CodeAnalysis.Sarif.Query.Evaluators;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool
+{
+    public class QueryCommandPropertyBagTests : IClassFixture<QueryCommandPropertyBagTests.TestFixture>
+    {
+        private const string FilePath = "property-bag-queries.sarif";
+
+        public class TestFixture
+        {
+            public TestFixture()
+            {
+                ResourceExtractor extractor = new ResourceExtractor(typeof(QueryCommandPropertyBagTests));
+                File.WriteAllText(FilePath, extractor.GetResourceText($"QueryCommand.{FilePath}"));
+            }
+        }
+
+        [Fact]
+        public void QueryCommand_CanAccessResultAndRulePropertyBags()
+        {
+            RunAndVerifyCount(2, "properties.name == 'Terisa'");
+            RunAndVerifyCount(2, "rule.properties.Category == 'security'");
+        }
+
+        [Fact]
+        public void QueryCommand_ComparesPropertyBagPropertyNamesCaseInsensitively()
```

Comment:
>QueryCommand_ComparesPropertyBagPropertyNamesCaseInsensitively [](start = 20, length = 62)

curious, why do we permit this? are our property bag keys case-insensitive?

Reply:
Our property bag keys are case-sensitive. Property "TeRisA" is indeed separate from "Terisa". All I can say is, "It seemed like a good idea at the time." I'll fix it.

---
In reply to: [496994367](https://github.com/microsoft/sarif-sdk/pull/2065#discussion_r496994367) [](ancestors = 496994367)

Reply:
Actually, I do remember -- I wanted to make it easier for a user to type the query without having to worry about the case of the property name. But still, I'll fix it.

---
In reply to: [497042412](https://github.com/microsoft/sarif-sdk/pull/2065#discussion_r497042412) [](ancestors = 497042412,496994367)